File: mifosng-android/src/main/java/com/mifos/services/data/ClientPayload.java
/*
 * This project is licensed under the open source MPL V2.
 * See https://github.com/openMF/android-client/blob/master/LICENSE.md
 */
package com.mifos.services.data;

/**
 * Created by ADMIN on 16-Jun-15.
 */
public class ClientPayload extends DefaultPayload {

    private String firstname;
    private String lastname;
    private String middlename;
    private int officeId;
    private boolean active;
    private String activationDate;
    private String submittedOnDate;
    private String dateOfBirth;
    private String mobileNo;
    private String externalId;
    //private int genderId;

    public String getFirstname() {
        return firstname;
    }

    public void setMiddlename(String mobilenumber) {
        this.middlename = middlename;
    }

    public String getMiddlename() {
        return middlename;
    }

    public String getExternalId() {
        return externalId;
    }

    public void setExternalId(String externalId) {
        this.externalId = externalId;
    }

    public void setFirstname(String firstname) {
        this.firstname = firstname;
    }

    public String getLastname() {
        return lastname;
    }

    public void setLastname(String lastname) {
        this.lastname = lastname;
    }

    public int getOfficeId() {
        return officeId;
    }

    public void setOfficeId(int officeId) {
        this.officeId = officeId;
    }

    public boolean isActive() {
        return active;
    }

    public void setActive(boolean active) {
        this.active = active;
    }

    public String getActivationDate() {
        return activationDate;
    }

    public void setActivationDate(String activationDate) {
        this.activationDate = activationDate;
    }

    public String getSubmittedOnDate() {
        return submittedOnDate;
    }

    public void setSubmittedOnDate(String submittedOnDate) {
        this.submittedOnDate = submittedOnDate;
    }

  /*  public Integer getGenderId() {
        return genderId;
    }

    public void setGenderId(Integer genderId) {
        this.genderId = genderId;
    }*/

    public String getDateOfBirth() {
        return dateOfBirth;
    }

    public void setDateOfBirth(String dateOfBirth) {
        this.dateOfBirth = dateOfBirth;
    }

    public String getMobileNo() {
        return mobileNo;
    }

    public void setMobileNo(String mobileNo) {
        this.mobileNo = mobileNo;
    }
}

File: core/database/src/main/java/com/mifos/core/objects/client/ClientPayload.kt
/*
 * Copyright 2024 Mifos Initiative
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 *
 * See https://github.com/openMF/android-client/blob/master/LICENSE.md
 */
package com.mifos.core.objects.client

import android.os.Parcelable
import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import com.mifos.core.database.MifosDatabase
import com.mifos.core.model.MifosBaseModel
import com.mifos.core.objects.noncore.DataTablePayload
import kotlinx.parcelize.Parcelize

/*
 * This project is licensed under the open source MPL V2.
 * See https://github.com/openMF/android-client/blob/master/LICENSE.md
 */
/**
 * Created by ADMIN on 16-Jun-15.
 */
@Parcelize
@Entity("ClientPayload")
data class ClientPayload(
    @PrimaryKey(autoGenerate = true)
    @Transient
    var id: Int? = null,

    @ColumnInfo("clientCreationTime")
    @Transient
    var clientCreationTime: Long? = null,

    @ColumnInfo("errorMessage")
    @Transient
    var errorMessage: String? = null,

    @ColumnInfo("firstname")
    var firstname: String? = null,

    @ColumnInfo("lastname")
    var lastname: String? = null,

    @ColumnInfo("middlename")
    var middlename: String? = null,

    @ColumnInfo("officeId")
    var officeId: Int? = null,

    @ColumnInfo("staffId")
    var staffId: Int? = null,

    @ColumnInfo("genderId")
    var genderId: Int? = null,

    @ColumnInfo("active")
    var active: Boolean? = null,

    @ColumnInfo("activationDate")
    var activationDate: String? = null,

    @ColumnInfo("submittedOnDate")
    var submittedOnDate: String? = null,

    @ColumnInfo("dateOfBirth")
    var dateOfBirth: String? = null,

    @ColumnInfo("mobileNo")
    var mobileNo: String? = null,

    @ColumnInfo("externalId")
    var externalId: String? = null,

    @ColumnInfo("clientTypeId")
    var clientTypeId: Int? = null,

    @ColumnInfo("clientClassificationId")
    var clientClassificationId: Int? = null,

    var address: List<Address>? = ArrayList(),

    @ColumnInfo("dateFormat")
    var dateFormat: String? = "dd MMMM YYYY",

    @ColumnInfo("locale")
    var locale: String? = "en",

    var datatables: List<DataTablePayload>? = null,
) : MifosBaseModel(), Parcelable
